torch/amp/grad_scaler.py

```diff
@@ -289,11 +275,15 @@ def _unscale_grads_(
 
             for device, per_dtype_grads in per_device_and_dtype_grads.items():
                 for grads in per_dtype_grads.values():
-                    torch._amp_foreach_non_finite_check_and_unscale_(
-                        grads,
-                        per_device_found_inf.get(device),
-                        per_device_inv_scale.get(device),
-                    )
+                    try:
+                        torch._amp_foreach_non_finite_check_and_unscale_(
+                            grads,
+                            per_device_found_inf.get(device),
+                            per_device_inv_scale.get(device),
+                        )
+                    except Exception as e:
+                        raise NotImplementedError(f"grad scaler required op function(torch._amp_foreach_non_finite_check_and_unscale_)"
+                                                  f"is not implemented on {device}")
```

Comment:
no, I'd prefer not to do this, suppressing the underlying exception can lead to some  pretty awful errors. E.g., if you had typoed the function name here, you'd still always hit the not implemnted case